code/game/machinery/bots/cprbot.dm

```diff
@@ -0,0 +1,501 @@
+		"I hope everyone is feeling alright today!",
+		"It's not magic, it's CPR Buddy 9000!",
+		"I should have been a plastic surgeon.",
+		"What kind of medbay is this? Everyone’s dropping like flies.",
+		"Each breath a day keeps me at bay!",
+		"I sense a disturbance in my circuit board, as if a million people stopped breathing and were suddenly silent."
+	)
+
+	/// Message to display when performing CPR
+	var/motivational_message = "Live! Live! Don't die on me now!"
+	/// List of patients who have been warned
+	var/list/has_said_to_patient = list()
+	/// Tracks the last time a message was spoken
+	var/last_message_time = 0
+
+/obj/structure/machinery/bot/cprbot/Initialize(mapload, ...)
+	. = ..()
+	start_processing()
+	playsound(loc, 'sound/CPRbot/CPRbot_poweron.ogg', 25, 1)
+	src.botcard = new /obj/item/card/id(src)
+	if(!LAZYLEN(src.botcard_access))
+		var/datum/job/Job = GLOB.RoleAuthority ? GLOB.RoleAuthority.roles_by_path[/datum/job/civilian/doctor] : new /datum/job/civilian/doctor
+		botcard.access = Job.get_access()
+	else
+		src.botcard.access = src.botcard_access
+
+/obj/structure/machinery/bot/cprbot/process()
+	if (health > 0)
+		think()
+
+		random_message()
+	else
+		stop_processing()
+
+/obj/structure/machinery/bot/cprbot/proc/think()
+	switch (state)
+		if (STATE_CPRBOT_IDLE)
+			find_and_move_to_patient()
+		if (STATE_CPRBOT_MOVING)
+			move_to_target()
+		if (STATE_CPRBOT_CPR)
+			try_perform_cpr()
+		if (STATE_CPRBOT_FOLLOWING_OWNER)
+			follow_owner()
+
+/obj/structure/machinery/bot/cprbot/proc/follow_owner()
+	if (!owner || isnull(owner))
+		go_idle()
+		return
+
+	// Check if the owner is in range of the bot's view
+	if (!(owner in view(search_radius, src)))
+		go_idle()
+		return
+
+	// If the bot has a path, proceed with walking along it
+	if (length(path))
+		var/turf/next = path[1]
+		if (loc == next)
+			path -= next // Remove the turf from the path once reached
+
+		walk_to(src, next, 0, move_to_delay)
+		return
+
+	// If the owner is not in range, move towards them directly
+	if (!patient_in_range(owner))
+		// If the bot is already moving, just return
+		if (state == STATE_CPRBOT_MOVING)
+			return
+
+		// Set the state to moving and walk directly to the owner's location
+		state = STATE_CPRBOT_MOVING
+		walk_to(src, owner, 0, movement_delay)
+	else
+		go_idle()
+
+/obj/structure/machinery/bot/cprbot/start_processing()
+	START_PROCESSING(SSobj, src)
+
+/obj/structure/machinery/bot/cprbot/stop_processing()
+	if (fast_processing)
+		STOP_PROCESSING(SSfastobj, src)
+	else
+		STOP_PROCESSING(SSobj, src)
+
+/obj/structure/machinery/bot/cprbot/Destroy()
+	human = null
+	path = null
+	botcard_access = null
+
+	stop_processing()
+	return ..()
+
+/obj/structure/machinery/bot/cprbot/proc/random_message()
+	if (!COOLDOWN_FINISHED(src, message_cooldown))
+		return // Exit if the cooldown period has not elapsed yet
+
+	// Send a message based on the current state
+	if (currently_healing)
+		speak(motivational_message)
```

Comment:
either add more motivational messages or add a prob(50) to not send it, it gets pretty spammy.